presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -600,6 +601,18 @@ public static long random(@SqlType(StandardTypes.BIGINT) long value)
         return ThreadLocalRandom.current().nextLong(value);
     }
 
+    @Description("a pseudo-random number from a normal distribution with mean, std, and p")
+    @ScalarFunction(alias = "rand_normal", deterministic = false)
+    @SqlType(StandardTypes.DOUBLE)
+    public static double randomNormal(@SqlNullable @SqlType(StandardTypes.DOUBLE) Double mean, @SqlNullable @SqlType(StandardTypes.DOUBLE) Double sd, @SqlNullable @SqlType(StandardTypes.DOUBLE) Double p)
```

Comment:
Remove `@SqlNullable` and make these primitive `double` instead of boxed `Double`. That will assure the engine never calls the function if any argument is null, and will perform better since it avoids boxing the argument values.